SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialClassInfoManager.cpp

```diff
@@ -292,6 +292,51 @@ UClass* USpatialClassInfoManager::GetClassByComponentId(Worker_ComponentId Compo
 	return nullptr;
 }
 
+uint32 USpatialClassInfoManager::GetComponentIdForClass(const UClass& Class) const
+{
+	const FString ClassPath = Class.GetPathName();
+	if (const FSchemaData* SchemaData = SchemaDatabase->ClassPathToSchema.Find(ClassPath))
+	{
+		return SchemaData->SchemaComponents[SCHEMA_Data];
+	}
+	return SpatialConstants::INVALID_COMPONENT_ID;
+}
+
+TArray<Worker_ComponentId> USpatialClassInfoManager::GetComponentIdsForClass(const UClass& BaseClass, const bool bIncludeDerivedTypes /* = true */) const
+{
+	TArray<Worker_ComponentId> OutComponentIds;
+
+	check(SchemaDatabase);
+	if (bIncludeDerivedTypes)
+	{
+		for (TObjectIterator<UClass> It; It; ++It)
```

Comment:
Looks expensive. You think it's worth caching the result?

Reply:
Maybe. Although I'd want to have perf data to prove the value. And after dealing with hot reload for caching the net cull distance stuff I'm reluctant to add that complexity for a negligible gain.

Reply:
Actually.... maybe the hot reload issues would be solved more easily by only caching when not running a PIE session?

Reply:
https://improbableio.atlassian.net/browse/UNR-1649